ballerina/tests/from_json_string_test.bal

```diff
@@ -0,0 +1,1072 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+// Possitive tests for fromJsonStringWithType() function.
+
+@test:Config
+isolated function testJsonStringToBasicTypes() returns error? {
+    int val1 = check fromJsonStringWithType("5");
+    test:assertEquals(val1, 5);
+
+    float val2 = check fromJsonStringWithType("5.5");
+    test:assertEquals(val2, 5.5);
```

Comment:
Can't we use data providers instead? Will enable the rest of the tests to run even if one fails.